data/mods/Magiclysm/traits/attunements.json

```diff
@@ -1128,7 +1128,7 @@
       "SOULFIRE",
       "WITHER_MAGE"
     ],
-    "flags": [ "ATTUNEMENT" ]
+    "flags": [ "ATTUNEMENT", "CRAFT_IN_DARKNESS", "READ_IN_DARKNESS" ]
```

Comment:
Maybe it’s worth splitting them up theatrically. Void Mage can see in perfect darkness, and then studying their spells in lightless chambers need underground is thematically resonant, but the void is a force of destruction and emptiness, not creation. Maybe Artificers should be able to craft in darkness as they start work and the divine madness strikes them and produces masterpieces. 

Or maybe Earth Elemental should because they’re one with the stone, shaping it in dark caverns. Or maybe they both should. My main point is that when we’re talking about magic we don’t need to stick strictly to logic to determine how things work.

Reply:
Removed darkness crafting from void mage and gave to artificer